internal/charmstore/store.go

```diff
@@ -505,6 +505,84 @@ func (s *Store) UpdateBaseEntity(url *charm.Reference, update interface{}) error
 	return nil
 }
 
+// Promulgate sets the base entity of url to be promulgated, and unsets promulgated
+// on any other base entity for charms with the same name. It also calculates the next
+// promulgated URL for the charms owned by the new owner and sets those charms.
+//
+// Note: This code is known to have some unfortunate (but not dangerous) race
+// conditions, it is considered that the promulgate operation will be significantly rare
+// that it is unlikely thay will occur.
+func (s *Store) Promulgate(url *charm.Reference) error {
+	entity, err := s.FindEntity(url)
+	if err != nil {
+		return errgo.Mask(err, errgo.Any)
+	}
+	bURL := baseURL(entity.URL)
```

Comment:
Let's have some comments explaining what's going on here, please.

e.g.
    // Set the promulgated flag on the base entity and at the same time
    // unset the promulgated flag on any other base entities with the same name.

?